spec/models/ont/run_factory_spec.rb

```diff
@@ -0,0 +1,178 @@
+require 'rails_helper'
+
+RSpec.describe Ont::RunFactory, type: :model, ont: true do
+  let!(:instrument_name) { 'test instrument name' }
+
+  before do
+    allow_any_instance_of(Pipelines::ConstantsAccessor).to receive(:instrument_name).and_return(instrument_name)
+  end
+
+  context '#initialise' do
+    it 'is invalid if run is invalid' do
+      errors = ActiveModel::Errors.new(Ont::Run.new)
+      errors.add('run', message: 'This is a test error')
+      allow_any_instance_of(Ont::Run).to receive(:valid?).and_return(false)
+      allow_any_instance_of(Ont::Run).to receive(:errors).and_return(errors)
+      factory = Ont::RunFactory.new([])
+      expect(factory).to_not be_valid
+    end
+
+    it 'is invalid if any flowcell is invalid' do
+      errors = ActiveModel::Errors.new(Ont::Flowcell.new)
+      errors.add('run', message: 'This is a test error')
+      allow_any_instance_of(Ont::Flowcell).to receive(:valid?).and_return(false)
+      allow_any_instance_of(Ont::Flowcell).to receive(:errors).and_return(errors)
+      factory = Ont::RunFactory.new(['PLATE-2-1234-2'])
+      expect(factory).to_not be_valid
+    end
+  end
+
+  context '#save' do
+    context 'valid build' do
+      context 'with no library names' do
+        it 'creates a run' do
```

Comment:
Should it still create a run if no library names are provided. I would have thought the flowcell validation would fail as it doesnt have a library?

Reply:
I don't know .. what does the UI send?  I can see why it doesn't harm to do this as well as it being a problem.  If the UI sends the run to the backend before the user has chosen libraries then updates the run as they drag in each library, then we probably do want the behaviour here.  If it only sends the run once the user has added some libraries then we could make the run factory reject the new run.  I didn't write this code, it just hadn't been merged back from the destination branch before you spotted this.  We should ask @Chris-Friend why he did it this way.

Reply:
I did this as I assumed a run was valid without any flowcells. It doesn't necessarily make sense in the domain, but in terms of code implementation and user workflows, as Stuart suggests, it gives us more flexibility on how runs are created